### PR TITLE
provider authentication: Ensure we validate provider project

### PR DESCRIPTION
**Pull request by stack72**
Fixes: #565

```
Diagnostics:
  google-native:storage/v1:Bucket (my-bucket):
    error: unable to find a valid Project. Set the Project by using:

     	 • `pulumi config set google-native:project my-project` or
     	 • Environment variables:`GOOGLE_PROJECT`, `GOOGLE_CLOUD_PROJECT`, `GCLOUD_PROJECT` or `CLOUDSDK_CORE_PROJECT`
```

### CODE REVIEW

File: provider/go.mod [hunk 1]
```diff
@@ -127,6 +128,7 @@ require (
 	github.com/rjeczalik/notify v0.9.2 // indirect
 	github.com/rogpeppe/go-internal v1.8.1 // indirect
 	github.com/ryanuber/go-glob v1.0.0 // indirect
+	github.com/ryboe/q v1.0.17 // indirect
```
Comment (lblackstone) [inline provider/go.mod:131]: I'm guessing this was just for testing?

**Review by viveklak — APPROVED** [review]

File: provider/go.mod (hunk 1 above)
Comment (viveklak) [inline provider/go.mod:131]: Not sure we need this? Sounds like we should be able to remove?

**Comment by viveklak** [comment]
I think #565 might be additionally effected by missing required field on the project? Can you try creating a bigquery dataset to make sure? Either way this PR is helpful, thanks!

File: provider/go.mod (hunk 1 above)
Comment (stack72) [reply provider/go.mod:131]: crap! Will remove :)

**Comment by stack72** [comment]
> I think #565 might be additionally effected by missing required field on the project? Can you try creating a bigquery dataset to make sure? Either way this PR is helpful, thanks!

Will follow up with a further test here :)

**Pull request merged**